dist/yoast-seo.js

```diff
@@ -2197,12 +2206,14 @@ YoastSEO.SnippetPreview.prototype.formatKeyword = function( textString ) {
 /**
  * formats the keyword for use in the URL by accepting - and _ in stead of space and by adding
  * <strong>-tags
+ * strips unwanted characters that could break the regex or give unwanted results
  *
  * @param textString
  * @returns {XML|string|void}
  */
 YoastSEO.SnippetPreview.prototype.formatKeywordUrl = function( textString ) {
-	var replacer = this.refObj.rawData.keyword.replace( " ", "[-_]" );
+	var keyword = this.refObj.stringHelper.cleanRegex( this.refObj.rawData.keyword );
+	var replacer = keyword.replace( " ", "[-_]" );
```

Comment:
What is a replacer? This is a keyword, and it's sanitized so maybe something like `sanitizedKeyword` will do. @atimmer please rename accordingly.